magali/_io.py

```diff
@@ -35,26 +35,93 @@ def read_qdm_harvard(path):
         coordinates are in µm and magnetic field in nT.
     """
     contents = scipy.io.loadmat(path)
+    coordinates, data_names, bz = _set_qdm_data_grid(contents)
+    data = _create_grid_qdm_data(coordinates, data_names, bz, path)
+
+    return data
+
+
+def _set_qdm_data_grid(contents):
```

Comment:
How about `_extract_data_qdm_harvard`? It's not really "setting" the data but extracting it from the `contents` dict.